lib/dry/system/container.rb

```diff
@@ -216,7 +227,15 @@ def import(other)
         # @return [self]
         #
         # @api public
-        def boot(name, opts = {}, &block)
+        def boot(name, when_condition: nil, **opts, &block)
+          if when_condition
+            begin
+              return unless when_condition.call
```

Comment:
@davydovanton here, instead of `when_condition.call`, we'd do:

``` ruby
case when_condition
  when Symbol then __send__(when_condition)
  when Proc then when_condition.call
  else
     raise "OOOOPS"
  end
```

Reply:
got it, thanks!